areas/extension/tests/AzureMcp.Extension.UnitTests/AzdCommandTest.cs

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine.Parsing;
+using System.Runtime.InteropServices;
+using AzureMcp.Extension.Commands;
+using AzureMcp.Core.Models.Command;
+using AzureMcp.Core.Services.ProcessExecution;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using Xunit;
+
+namespace AzureMcp.Tests.Areas.Extension.UnitTests;
+
+[Trait("Area", "Extension")]
+public sealed class AzdCommandTests
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly IExternalProcessService _processService;
+    private readonly ILogger<AzdCommand> _logger;
+
+    private const int TempDirSuffixLength = 8;
+
+    public AzdCommandTests()
+    {
+        _processService = Substitute.For<IExternalProcessService>();
+        _logger = Substitute.For<ILogger<AzdCommand>>();
+
+        var collection = new ServiceCollection();
+        collection.AddSingleton(_processService);
+        _serviceProvider = collection.BuildServiceProvider();
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_ReturnsSuccessResult_WhenCommandExecutesSuccessfully()
+    {
+        // Arrange
+        var command = new AzdCommand(_logger);
+        var parser = new Parser(command.GetCommand());
+        var args = parser.Parse("--command \"env list\" --cwd test-dir");
+        var context = new CommandContext(_serviceProvider);
+
+        var expectedOutput = "env1\nenv2";
+
+        _processService.ExecuteAsync(
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<int>())
+            .Returns(new ProcessResult(0, expectedOutput, string.Empty, "azd env list --cwd test-dir --no-prompt"));
```

Comment:
The azd command can take 4 parameters. Even if the last one is nullable, I think it's better we include it in any calls we make in this test class.
```suggestion
        _processService.ExecuteAsync(
            Arg.Any<string>(),
            Arg.Any<string>(),
            Arg.Any<int>(),
            Arg.Any<IEnumerable<string>>()
            .Returns(new ProcessResult(0, expectedOutput, string.Empty, "azd env list --cwd test-dir --no-prompt"));
```